### PR TITLE
Updated harbor and keycloak provider config to point to Terraform registry

### PR DESCRIPTION
**Pull request by gesparza3**
This PR updates the required_providers config to point to the newly published `harbor` and `keycloak` providers.

### CODE REVIEW

**Comment by mrparkers** [comment]
can you help me understand the context behind the "tf upgrade todo" comments?

**Comment by gesparza3** [comment]
> can you help me understand the context behind the "tf upgrade todo" comments?

This was generated from running `terrafrom 0.13upgrade`, I've pushed up a change that cleaned up those comments.

**Comment by gesparza3** [comment]
This should now be good to review, I've gotten the plan working after moving providers over to `environment-aws`. The one change I don't like is moving the `harbor_admin_password` generation to `environment-aws` but this was necessary as I needed the provider to be configured with the password prior to instantiating the Harbor module now that it's been pulled out of toolchain. This should be able to be resolved as a part of creating the Harbor modules, since we can set it as an output variable so it can be set on the Harbor provider before applying the configuration.

**Comment by mrparkers** [comment]
> The one change I don't like is moving the harbor_admin_password generation to environment-aws but this was necessary as I needed the provider to be configured with the password prior to instantiating the Harbor module now that it's been pulled out of toolchain.

What do you think about generating this password one time and storing it in Vault?  Then we can pull it in with the `vault_generic_secret` data source.

**Comment by gesparza3** [comment]
That sounds fine to me, we do it in other places like for the `keycloak_admin_password` so it makes sense to do it here too.

**Comment by gesparza3** [comment]
Updated to retrieve `harbor_admin_password` from Vault.

File: stacks/environment-aws/lead.tf [hunk 1]
```diff
@@ -33,6 +37,7 @@ module "toolchain" {
   google_identity_provider_client_secret = var.enable_google_login ? data.vault_generic_secret.keycloak.data["google-idp-client-secret"] : ""
   enable_test_user                       = var.enable_test_user
   test_user_password                     = var.enable_test_user ? data.vault_generic_secret.keycloak.data["test-user-password"] : ""
+  harbor_admin_password                  = data.vault_generic_secret.keycloak.data["admin-password"]
```
Comment (mrparkers) [inline stacks/environment-aws/lead.tf:40]:
```suggestion
  harbor_admin_password                  = data.vault_generic_secret.harbor.data["admin-password"]
```

**Comment by gesparza3** [comment]
Made that fix. I don't know how I missed that, sorry about that.

**Pull request merged**